### PR TITLE
Use frameworkHandle to get listers

### PR DESCRIPTION
**Pull request by mrkm4ntr**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Use frameworkHandle to get listers in each plugin.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/nodevolumelimits/node_volume_limits.go [hunk 1]
```diff
@@ -48,9 +49,21 @@ func (pl *NodeVolumeLimits) Filter(ctx context.Context, _ *framework.CycleState,
 	return migration.PredicateResultToFrameworkStatus(reasons, err)
 }
 
-// New initializes a new plugin and returns it.
-func New(csiNodeInfo predicates.CSINodeInfo, pvInfo predicates.PersistentVolumeInfo, pvcInfo predicates.PersistentVolumeClaimInfo, classInfo predicates.StorageClassInfo) framework.Plugin {
-	return &NodeVolumeLimits{
-		predicate: predicates.NewCSIMaxVolumeLimitPredicate(csiNodeInfo, pvInfo, pvcInfo, classInfo),
+// NewFactory returns function that initializes a new plugin and returns it.
+func NewFactory(csiNodeInfo predicates.CSINodeInfo) framework.PluginFactory {
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/nodevolumelimits/node_volume_limits.go]: just name it ```New```

File: pkg/scheduler/framework/plugins/volumezone/volume_zone_test.go [hunk 2]
```diff
@@ -47,6 +47,12 @@ func createPodWithVolume(pod, pv, pvc string) *v1.Pod {
 	}
 }
 
+func createVolumeZone(pvInfo predicates.PersistentVolumeInfo, pvcInfo predicates.PersistentVolumeClaimInfo, classInfo predicates.StorageClassInfo) framework.Plugin {
+	return &VolumeZone{
+		predicate: predicates.NewVolumeZonePredicate(pvInfo, pvcInfo, classInfo),
+	}
+}
+
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/volumezone/volume_zone_test.go]: we don't need this function

File: pkg/scheduler/framework/plugins/volumezone/volume_zone_test.go [hunk 3]
```diff
@@ -154,7 +160,7 @@ func TestSingleZone(t *testing.T) {
 		t.Run(test.name, func(t *testing.T) {
 			node := &schedulernodeinfo.NodeInfo{}
 			node.SetNode(test.Node)
-			p := New(pvInfo, pvcInfo, nil)
+			p := createVolumeZone(pvInfo, pvcInfo, nil)
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/volumezone/volume_zone_test.go]: just do this
```p :=  &VolumeZone{
		predicate: predicates.NewVolumeZonePredicate(pvInfo, pvcInfo, nil),
	}
```

and below, remove the cast:
```gotStatus := p.Filter(context.Background(), nil, test.Pod, node)```

File: pkg/scheduler/framework/plugins/nodevolumelimits/node_volume_limits_test.go [hunk 4]
```diff
@@ -47,6 +47,12 @@ const (
 	hostpathInTreePluginName = "kubernetes.io/hostpath"
 )
 
+func createNodeVolumeLimits(csiNodeInfo predicates.CSINodeInfo, pvInfo predicates.PersistentVolumeInfo, pvcInfo predicates.PersistentVolumeClaimInfo, classInfo predicates.StorageClassInfo) framework.Plugin {
+	return &NodeVolumeLimits{
+		predicate: predicates.NewCSIMaxVolumeLimitPredicate(csiNodeInfo, pvInfo, pvcInfo, classInfo),
+	}
+}
+
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/nodevolumelimits/node_volume_limits_test.go]: please remove this func

File: pkg/scheduler/framework/plugins/nodevolumelimits/node_volume_limits_test.go [hunk 5]
```diff
@@ -444,7 +450,7 @@ func TestNodeVolumeLimits(t *testing.T) {
 				defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigrationAWS, false)()
 			}
 
-			p := New(getFakeCSINodeInfo(csiNode),
+			p := createNodeVolumeLimits(getFakeCSINodeInfo(csiNode),
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/nodevolumelimits/node_volume_limits_test.go]: do this:

```
p := &NodeVolumeLimits{
		predicate: predicates.NewCSIMaxVolumeLimitPredicate(getFakeCSINodeInfo(csiNode), getFakeCSIPVInfo(test.filterName, test.driverNames...), getFakeCSIPVCInfo(test.filterName, "csi-sc", test.driverNames...), getFakeCSIStorageClassInfo("csi-sc", test.driverNames[0])),
	}
```

and remove the cast to FilterPlugin below

**Comment by mrkm4ntr** [comment]
Resolved conflict again.

**Pull request merged**